Plugins/Autodiff/include/Acts/Plugins/Autodiff/GenericDefaultExtension.hpp

```diff
@@ -0,0 +1,241 @@
+// This file is part of the Acts project.
+//
+// Copyright (C) 2018-2020 CERN for the benefit of the Acts project
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+/// @warning This file is just copied from the Core/Propagation directory,
+/// and then templated. It was last updated in July 2020.
+
+#pragma once
+
+#include "Acts/Utilities/Helpers.hpp"
+
+namespace Acts {
+
+/// @brief Default evaluater of the k_i's and elements of the transport matrix
+/// D of the RKN4 stepping. This is a pure implementation by textbook.
+/// @note This is the templated generic type. The default one with 'double'
+/// is a typedef below.
+template <typename float_t>
+struct GenericDefaultExtension {
+  /// @brief Default constructor
+  GenericDefaultExtension() = default;
```

Comment:
Do I understand correctly that this is a copy of the core default extension, where you 'only' add a floating point type, so you can plug in the special types from autodiff.

Would there be a way to not have to copy the class(es) here? Do you make other changes to the class other than the template parameter?

Reply:
Yeah, I also thought about just changing the files in the Core modules, but there are some changes that are I think not common in rest of ACTS:
- I use custom typedefs for Eigen types, like:
```c++
  template <typename T>
  using Vector3D = Eigen::Matrix<T, 3, 1>;
```
- To support some function which rely on `float` (like `computeEnergyLossMean(...)`) I had to do some a bit unusual things like `static_cast<float>(static_cast<double>(qop))`.

So these changes would bring a few inconsitencies with the rest of ACTS into this... But in principle it should be no problem to change the original files.